test/sanity/validate-modules/main.py

```diff
@@ -1394,12 +1394,14 @@ def _check_for_new_args(self, doc, metadata):
                     if existing_version:
                         break
                 current_version = details.get('version_added')
-                if current_version != existing_version:
+
+                # We expect the added_version to be a string, not a float
+                if current_version and existing_version and current_version != str(existing_version):
```

Comment:
I don't think this verbose logic is necessary.

Just use

```
if str(current_version) != str(existing_version):
```

Reply:
Also, if we are trying to enforce string types, that needs to be addressed in the schema also.

Right now we have:

```
version_added': Any(float, *string_types),
```

So a `float` is a legal value currently.

Reply:
>     if str(current_version) != str(existing_version):

We deliberately want to compare the current version to a string of the existing. And that won't work for None values. If you make both a string like you do, you allow people making a string a float and it won't be covered by these tests.

PS This was my first implementation, but didn't fit what we needed.

Reply:
> Also, if we are trying to enforce string types, that needs to be addressed in the schema also.

I don't want to force everybody to make it a string per se. But if people introduce new `version_added` as a float, or modifying it to be a float, we give an error.

And if people are fixing existing floats to become a string, we would allow it without giving an issue. (See https://github.com/ansible/ansible/pull/52493)